include/boost/multiprecision/detail/et_ops.hpp

```diff
@@ -635,7 +828,18 @@ operator*(const detail::expression<tag, Arg1, Arg2, Arg3, Arg4>& a, const number
 }
 
 template <class tag, class Arg1, class Arg2, class Arg3, class Arg4, class B, expression_template_option ET>
-inline BOOST_MP_CXX14_CONSTEXPR typename detail::expression<detail::multiplies, detail::expression<tag, Arg1, Arg2, Arg3, Arg4>, number<B, ET> >::result_type
+inline BOOST_MP_CXX14_CONSTEXPR typename std::enable_if<
+   std::is_same<typename detail::expression<detail::multiplies, detail::expression<tag, Arg1, Arg2, Arg3, Arg4>, number<B, ET> >::result_type, number<B, ET>>::value,
+   typename detail::expression<detail::multiplies, detail::expression<tag, Arg1, Arg2, Arg3, Arg4>, number<B, ET> >::result_type>::type
+operator*(const detail::expression<tag, Arg1, Arg2, Arg3, Arg4>& a, number<B, ET>&& b)
+{
+   b *= a;
+   return static_cast<number<B, et_on>&&>(b);
```

Comment:
Seeing the CI results at https://drone.cpp.al/boostorg/multiprecision/292/179/2, can you comment on that `static_cast`? The type of `b` is `number<B, ET>&&`. Why should it cast nicely into `number<B, et_on>&&`?